internal/clients/nightfall/nightfall.go

```diff
@@ -262,19 +283,52 @@ func (n *Client) scanAllContent(ctx context.Context, logger logger.Logger, cts [
 }
 
 // Scan send /scan request to Nightfall API and return findings
-func (n *Client) Scan(ctx context.Context, logger logger.Logger, items []string) ([][]nightfallAPI.ScanResponse, error) {
+func (n *Client) Scan(
+	ctx context.Context,
+	logger logger.Logger,
+	items []string,
+) ([][]nightfallAPI.ScanResponse, error) {
 	APIKey := nightfallAPI.APIKey{
 		Key:    n.APIKey,
 		Prefix: "",
 	}
 	newCtx := context.WithValue(ctx, nightfallAPI.ContextAPIKey, APIKey)
 	request := n.createScanRequest(items)
-	resp, _, err := n.APIClient.ScanAPI().ScanPayload(newCtx, request)
-	if err != nil {
-		logger.Error(fmt.Sprintf("Error from Nightfall API, unable to successfully scan %d items", len(items)))
-		return nil, err
+	return n.makeScanRequest(newCtx, logger, request)
+}
+
+func (n *Client) makeScanRequest(
+	ctx context.Context,
+	logger logger.Logger,
+	request nightfallAPI.ScanRequest,
+) ([][]nightfallAPI.ScanResponse, error) {
+	delay := n.InitialRetryDelay
+	for i := 0; i < maxRetries; i++ {
+		resp, httpResp, err := n.APIClient.ScanPayload(ctx, request)
+		if err != nil {
+			if httpResp.StatusCode == http.StatusTooManyRequests {
```

Comment:
just confirming - I know you had mentioned receiving 500 status codes too. Is it possible to get any other status codes when we're being rate limited or should this 1 cover it?

Reply:
Nope this is the only status code we receive when we get rate limited from the API Platform. 

When the API Platform gets rate limit errors from the Amazon API Gateway, those get masked as 500s, so I'm not too sure if we want to include 500s in this condition, which may also re-attempt making the request on other unrelated 500 errors.

Reply:
ah I see. So if the API Platform rate limits us, we get that status code. If there are rate limit issues b/w the API Platform (i.e. Lambda) and the Gateway (i.e. Metadata service holding the companyID's), we get a 500. Makes sense!